### PR TITLE
tidb-in-kubernetes: update GCP GKE deployment docs

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? 

- Add missing content related to GKE Terraform
- Update GCP Terraform modules

### What is the related PR or file link(s)? 

- https://github.com/pingcap/docs/pull/1411
- https://github.com/pingcap/docs/pull/1424

### Which version does your change affect? 

`dev`, `v3.0`

### CODE REVIEW

**Comment by lilin90** [comment]
@DanielZhangQD @aylei Please take a technical review carefully because many descriptions in the two pingcap/docs PRs are confusing and not clear enough. Thanks!

File: dev/tidb-in-kubernetes/deploy/gcp-gke.md [hunk 1]
```diff
@@ -323,8 +434,198 @@ gcloud compute instance-groups managed delete-instances gke-my-cluster-monitor-p
...
+
 如果你不再需要之前的数据，执行完 `terraform destroy` 后，可通过 Google Cloud 控制台或者 `gcloud` 删除磁盘。
 
+如果你不再需要之前的数据，并且想要删除正在使用的磁盘，有以下两种方法可以完成此操作：
+
+- 手动删除：在 Google Cloud Console 中删除磁盘，或使用 `gcloud` 命令行工具执行删除操作。
+
+- 在执行 `terraform destroy` 之前将 Kubernetes 的 PV (Persistent Volume) 回收策略设置为 `Delete`，具体操作为在 `terraform destroy` 之前运行以下 `kubectl`命令：
+
+    ```bash
```
Comment (aylei) [inline dev/tidb-in-kubernetes/deploy/gcp-gke.md]: Add copyable macro

File: dev/tidb-in-kubernetes/deploy/gcp-gke.md [hunk 2]
```diff
@@ -323,8 +434,198 @@ gcloud compute instance-groups managed delete-instances gke-my-cluster-monitor-p
...
+    ```
+
+    上述命令将获取 TiDB 集群命名空间中的 PVC (Persistent Volume Claim)，并将 PV 的回收策略设置为 `Delete`。在执行 `terraform destroy` 过程中删除 PVC 时，也会将磁盘删除。
+
+    下面是一个名为 `change-pv-reclaimpolicy.sh` 的脚本。相对于仓库根目录来说，它在 `deploy/gcp` 中简化了上述过程。
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    ./change-pv-reclaimpolicy.sh /path/to/kubeconfig/file tidb-cluster-namespace
```
Comment (aylei) [inline dev/tidb-in-kubernetes/deploy/gcp-gke.md]:
```suggestion
    ./change-pv-reclaimpolicy.sh /path/to/kubeconfig/file <tidb-cluster-namespace>
```

File: dev/tidb-in-kubernetes/deploy/gcp-gke.md [hunk 3]
```diff
@@ -323,8 +434,198 @@ gcloud compute instance-groups managed delete-instances gke-my-cluster-monitor-p
...
+2. 根据具体需求，使用 Terraform 脚本将上述模块进行组合。
+
+如果采用了最佳实践，集群中的 Terraform 状态不会相互干扰，并且扩展起来很方便。示例如下：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+# 假设已在项目根目录
+mkdir -p deploy/gcp-staging
+vim deploy/gccp-staging/main.tf
```
Comment (DanielZhangQD) [inline dev/tidb-in-kubernetes/deploy/gcp-gke.md]: gccp - > gcp

Comment (lilin90) [reply dev/tidb-in-kubernetes/deploy/gcp-gke.md]: @DanielZhangQD Sharp eyes! 👍

**Comment by lilin90** [comment]
@aylei @DanielZhangQD Thanks! Comments addressed. PTAL again~

File: dev/tidb-in-kubernetes/deploy/gcp-gke.md [hunk 4]
```diff
@@ -227,17 +259,72 @@ Check Table Before Drop: false
...
+  tikv_node_count            = 2
+  tidb_node_count            = 1
+  monitor_node_count         = 1
+}
+```
+
+> **注意：**
+>
+> - 每个集群的 `cluster_name` 必须是唯一的。
+> - 创建的实际节点数乘以该 Region 中的可用区数。
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/deploy/gcp-gke.md]: This sentence does not make sense to me...

File: dev/tidb-in-kubernetes/deploy/gcp-gke.md [hunk 5]
```diff
@@ -323,8 +434,198 @@ gcloud compute instance-groups managed delete-instances gke-my-cluster-monitor-p
...
+
+module "bastion" {
+  source             = "../modules/gcp/bastion"
+  vpc_name           = module.vpc.vpc_name
+  public_subnet_name = module.vpc.public_subnetwork_name
+  gcp_project        = var.GCP_PROJECT
+  bastion_name       = "${locals.gke_name}-tidb-bastion"
+}
+
+# HACK: enforces Helm to depend on the GKE cluster
```
Comment (dcalvin) [inline dev/tidb-in-kubernetes/deploy/gcp-gke.md]:
```suggestion
# HACK: 强制使 Helm 依赖 GKE 集群
```

File: dev/tidb-in-kubernetes/deploy/gcp-gke.md (hunk 4 above)
Comment (lilin90) [reply dev/tidb-in-kubernetes/deploy/gcp-gke.md]: @dcalvin  Me too... Many confusing expressions in that English version. The English version for this sentence is `The actual number of nodes created are multiplied by the number of availability zones in the region.`.

Comment (lilin90) [reply dev/tidb-in-kubernetes/deploy/gcp-gke.md]: @aylei @DanielZhangQD Could you help explain the meaning here? Thanks!

Comment (lilin90) [reply dev/tidb-in-kubernetes/deploy/gcp-gke.md]: @dcalvin I've confirmed with @DanielZhangQD. That sentence means "实际的总节点数等于配置文件中的节点数乘以可用区的个数".

Comment (lilin90) [reply dev/tidb-in-kubernetes/deploy/gcp-gke.md]: Updated.

**Comment by lilin90** [comment]
@dcalvin Thanks! All comments addressed. PTAL again~

File: dev/tidb-in-kubernetes/deploy/gcp-gke.md (hunk 4 above)
Comment (dcalvin) [reply dev/tidb-in-kubernetes/deploy/gcp-gke.md]: We should also update in docs...

Comment (lilin90) [reply dev/tidb-in-kubernetes/deploy/gcp-gke.md]: Yes. That document in pingcap/docs needs a lot of updates: typo, format, wording, ....

**Pull request merged**